taxonomy/api.py

```diff
@@ -0,0 +1,245 @@
+    """
+    return Vocabulary.objects.get(id=vocabulary_id)
+
+
+# pylint: disable=too-many-arguments
+@transaction.atomic
+def create_vocabulary(
+        repository_id, name, description, required, vocabulary_type, weight
+):
+    """
+    Create a Vocabulary and save it in database
+
+    Args:
+        repository_id (int): Repository id
+        name (unicode): Name of Vocabulary
+        description (unicode): Description of Vocabulary
+        required (bool): Is Vocabulary required?
+        vocabulary_type (unicode): Vocabulary type
+        weight (int): Weight
+
+    Returns:
+        Vocabulary: The Vocabulary which was just added to the database
+
+    """
+
+    if vocabulary_type != "m" and vocabulary_type != "f":
+        raise TaxonomyAPIException("vocabulary_type must be either"
+                                   " 'm' (managed) or 'f' (free tagging)")
+
+    repository = Repository.objects.get(id=repository_id)
+
+    return Vocabulary.objects.create(
+        repository=repository,
+        name=name,
+        description=description,
+        required=required,
+        vocabulary_type=vocabulary_type,
+        weight=weight,
+    )
+
+
+def delete_vocabulary(vocabulary_id):
+    """
+    Delete Vocabulary with given id
+
+    Args:
+        vocabulary_id (int): Vocabulary id
+    """
+    Vocabulary.objects.filter(id=vocabulary_id).delete()
+
+
+def get_term(term_id):
+    """
+    Get Term with existing id
+
+    Args:
+        term_id (int): Term id
+
+    Returns:
+        Term: The Term with the id
+    """
+    return Term.objects.get(id=term_id)
+
+
+@transaction.atomic
+def create_term(vocabulary_id, label, weight):
+    """
+    Create new Term and save it in database
+
+    Args:
+        vocabulary_id (int): Vocabulary id
+        label (unicode): Term label
+        weight (int): Weight for term
+
+    Returns:
+        Term: The newly created Term
+    """
+    vocabulary = Vocabulary.objects.get(id=vocabulary_id)
+
+    return Term.objects.create(
+        vocabulary=vocabulary,
+        label=label,
+        weight=weight
+    )
+
+
+def delete_term(term_id):
+    """
+    Delete an existing Term
+
+    Args:
+        term_id (int): Term id
+    """
+    Term.objects.filter(id=term_id).delete()
+
+
+def get_learning_resources_for_term(term_id):
+    """
+    Get LearningResources using a particular Term
+
+    Args:
+        term_id (int): Term id
+
+    Returns:
+        sequence of LearningResource relating to existing Term
+    """
+    return Term.objects.get(id=term_id).learning_resources.order_by('id')
+
+
+def get_terms_for_learning_resource(learning_resource_id):
+    """
+    Get Terms using a particular LearningResource
+
+    Args:
+        learning_resource_id (int): Learning object id
+
+    Returns:
+        sequence of Term relating to existing LearningResource
+    """
+    return (LearningResource.objects.get(id=learning_resource_id).terms.
```

Comment:
It's more Pythonic not to use extra parens for wrapping, since you get that "for free" from the parens already in the line.

```python
return LearningResource.objects.get(
    id=learning_resource_id).terms.order_by('id')
```